bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -55,37 +64,33 @@ public class RemoteControllerWebSocket extends RemoteController implements Liste
     private final WebSocketArt webSocketArt;
     private final WebSocketV2 webSocketV2;
 
+    // refresh limit for current app update (in seconds)
+    private static final long UPDATE_CURRENT_APP_REFRESH = 10;
+    private Instant previousUpdateCurrentApp = Instant.ofEpochMilli(0);
+
     // JSON parser class. Also used by WebSocket handlers.
-    final Gson gson = new Gson();
+    public final Gson gson = new Gson();
 
     // Callback class. Also used by WebSocket handlers.
-    final RemoteControllerWebsocketCallback callback;
+    final RemoteControllerService callback;
 
     // Websocket client class shared by WebSocket handlers.
     final WebSocketClient client;
 
-    // temporary storage for source app. Will be used as value for the sourceApp channel when information is complete.
-    // Also used by Websocket handlers.
-    @Nullable
-    String currentSourceApp = null;
+    // App File servicce
+    private final SamsungTvAppWatchService samsungTvAppWatchService;
 
-    // last app in the apps list: used to detect when status information is complete in WebSocketV2.
-    @Nullable
-    String lastApp = null;
-
-    // timeout for status information search
-    private static final long UPDATE_CURRENT_APP_TIMEOUT = 5000;
-    private long previousUpdateCurrentApp = 0;
+    // list instaled apps after 2 updates
+    public int updateCount = 0;
 
     // UUID used for data exchange via websockets
     final UUID uuid = UUID.randomUUID();
 
     // Description of Apps
-    @NonNullByDefault()
-    class App {
-        String appId;
-        String name;
-        int type;
+    public class App {
+        public String appId;
+        public String name;
+        public int type;
```

Comment:
Could this be `enum` instead of `int` to clearly document the types? What does 2 mean? Example:

```
public enum AppType {
    @SerializedName("2")
    MEANINGFUL_NAME
}
```

Reply:
There is no good name for it, the eden launcher uses 2 to mean "DEEP_LINK" and 4 to mean "NATIVE_LAUNCH", but the websocket launcher uses 2 to mean "ms.application" and 4 to mean "ms.webapplication" - depending on the TV model year (eden launcher is removed as of 2019).

As there are only two valid values returned by the TV api (2 or 4) I use it as a boolean. and substitute the actual string depending on which service is launching the app.

like this:
```
params.data.action_type = deepLink ? "DEEP_LINK" : "NATIVE_LAUNCH";
```
or
```
String apptype = (appType) ? "ms.application.get" : "ms.webapplication.get";
```
Also, as of 2020, only type 2 apps seem to exist, Samsung seems to have dropped the type 4 apps.
I don't think an enum is really justified for a variable that only has two possible values.